### PR TITLE
julia: remove unsupported versions

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

We have many versions of julia that no longer receive upstream updates. julia 0.4 was also marked as broken start of this year.
As a result we cannot update libraries that break compatibility (https://github.com/NixOS/nixpkgs/pull/44137)
Also julia-git was removed since it was not updated in two years and older then current stable version.

### CODE REVIEW

**Comment by Mic92** [comment]
cc @rbvermaa @FRidh @7c6f434c

**Review by FRidh — APPROVED** [review]

File: pkgs/top-level/all-packages.nix
Comment (garrison) [inline pkgs/top-level/all-packages.nix]: Is there a reason the name `julia_06` was removed?  I have witnessed this oscillate over time and it is frustrating that these names come and go, especially since each minor version of julia before 1.0 is not backwards compatible with others.  At the moment, as of c745ed1f7c5398848c0361e75be3c2d82aaab95b, there is `julia`, `julia_07`, and `julia_10`, but no way of specifying julia 0.6 explicitly.

Comment (garrison) [reply pkgs/top-level/all-packages.nix]: As an aside, I am a pretty serious user (and also a developer of) julia, but only a casual user/developer of nixpkgs.  If having my perspective on julia-related pull requests here in the future would be helpful, feel free to add me to the maintainers list.  (I'm guessing this is the most reliable way to be CCed on any related changes, which would allow me to make comments before things are merged.)

Comment (7c6f434c) [reply pkgs/top-level/all-packages.nix]: I agree on both counts.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Julia 0.6 was not removed only julia_04 and julia_05.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: However I should have kept `julia_06` as an alias.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: This is already fixed: 0e4197b34bf5dd877817165fedcb9401f019d656

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: I backported this as well:

[detached HEAD 4a62ef08009] julia: readd julia_06 name, make julia an alias that selects a version, as we usually do for incompatible versions
 Author: Michael Raskin <7c6f434c@mail.ru>
 Date: Fri Sep 7 11:36:20 2018 +0300
 1 file changed, 3 insertions(+), 1 deletion(-)

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: I would appreciate if we have active julia maintainer. Then I would feel more comfortable to have multiple versions of julia in nixpkgs. At the time of doing this pull request I had the feeling that no take care of removing old versions, that's why I wanted to shift to having a single version only.

**Pull request merged**
